app/src/main/java/com/hover/stax/inapp_banner/BannerViewModel.kt

```diff
@@ -0,0 +1,27 @@
+package com.hover.stax.inapp_banner
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.hover.stax.database.DatabaseRepo
+import kotlinx.coroutines.launch
+
+class BannerViewModel(application: Application, repo: DatabaseRepo) : AndroidViewModel(application) {
+    private val qualifiedBannerLiveData: MutableLiveData<Banner> = MutableLiveData<Banner>()
+    private val bannerUtils = BannerUtils(getApplication())
+
+    init {
+            viewModelScope.launch {
+                val hasTransactionLastMonth: Boolean = repo.hasTransactionLastMonth()
+                qualifiedBannerLiveData.postValue(bannerUtils.getQualifiedBanner(hasTransactionLastMonth))
+            }
+    }
+
+    fun qualifiedBanner(): LiveData<Banner> = qualifiedBannerLiveData
```

Comment:
We can directly access the `qualifiedBannerLiveData` object so this is necessary.

Reply:
Directly accessing it counters a clean approach to getting results from LiveData. 

- Variables should be separate from its getter.

I understand kotlin does something similar for data classes, but it this case, kotlin under the hood generates getters and setter methods.